pkg/daemon/update.go

```diff
@@ -701,15 +712,25 @@ func (dn *Daemon) getPendingState() (string, error) {
 		return "", err
 	}
 	if len(journalOutput) == 0 {
-		return "", nil
+		if !loggerWithoutJournalOption() {
+			return "", nil
+		}
+		cmd := "journalctl -o cat | grep OLD_LOGGER"
```

Comment:
Eeek.  Scraping all journal messages for that?

How about we detect on *startup* whether the host is rhel7, and add that as a flag in the daemon struct.  That'd be useful for a lot of other things.

Reply:
we scrape all journal oly if --journald isn't supported so we need to do this anyway right? I didn't want to add a specific rhel7 flag as I was just testing for functionality which should be generally better. If rhel7.7 upgrades util linux to a logger that does have --journald, we'd be still using this awful ack if we just test RHEL versions

Reply:
> If rhel7.7 upgrades util linux to a logger that does have --journald, we'd be still using this awful ack if we just test RHEL versions

Even if it did upgrade which...seems *very* unlikely (right?) we'd still have an upgrade hazard where for previous versions we'd write one way and then later the other way.  Seems better to just always live with one format.

Reply:
> Eeek. Scraping all journal messages for that?

to reply specifically on that. With this patch, we're not _always_ scraping all journal, `if !loggerWithoutJournalOption() {` takes care of telling us if we need to do that in case `logger` doesn't have `--journald`.

Reply:
> Even if it did upgrade which...seems very unlikely (right?) we'd still have an upgrade hazard where for previous versions we'd write one way and then later the other way. Seems better to just always live with one format.

Yeah, agree it's very unlikely indeed, but since we don't need to play with a rhel7 flag today, isn't testing for the logger functionality in itself enough? (I'm super happy to change it anyway)

Reply:
`OLD_LOGGER` seems a bit too generic...while it's unlikely, that string could appear in log messages from any othe process.

(Ideally...we'd have the MCD logging under `machine-config-daemon.service` that we create on the host.  That's another topic)

How about calling it `OPENSHIFT_MACHINE_CONFIG_DAEMON_LEGACY_LOG_HACK` ? :wink:

Reply:
> How about calling it `OPENSHIFT_MACHINE_CONFIG_DAEMON_LEGACY_LOG_HACK` ?

:joy: I'll for it, that's a good one indeed to avoid collisions